ros/src/twist_controller/dbw_node.py

```diff
@@ -7,136 +7,133 @@
+                throttle, brake, steer = self._controller.control(self._twist_cmd, self._current_velocity)
+            elif not self._dbw_enabled:
+                # reset pid controller
+                self._controller.reset()
+	    elif not self._tf_init_done:
+		# brake until tensorflow is initialized
+                throttle, brake, steer = 0.0, BrakeCmd.TORQUE_BOO * 2.0, 0.0
+
+            self._publish(throttle, brake, steer)
             rate.sleep()
 
-    def dbw_enabled_cb(self, msg):
-        self.dbw_enabled = msg
-
-    def current_velocity_cb(self, msg):
-        self.current_velocity = msg
-
-    def twist_cmd_cb(self, msg):
-        self.twist_cmd = msg
-
-    def publish(self, throttle, brake, steer):
-        if abs(self.last_throttle - throttle) > EPSILON_THROTTLE:
-            self.last_throttle = throttle
-            tcmd = ThrottleCmd()
-            tcmd.enable = True
-            tcmd.pedal_cmd_type = ThrottleCmd.CMD_PERCENT
-            tcmd.pedal_cmd = throttle
-            self.throttle_pub.publish(tcmd)
-            rospy.loginfo("Issued throttle command, value={}".format(throttle))
-        else:
-            rospy.logdebug(
-                "Did no issue throttle command, value={}, last value={}".format(throttle, self.last_throttle))
-
-        if abs(self.last_steer - steer) > EPSILON_STEER:
-            self.last_steer = steer
-            scmd = SteeringCmd()
-            scmd.enable = True
-            scmd.steering_wheel_angle_cmd = steer
-            self.steer_pub.publish(scmd)
-            rospy.loginfo("Issued steer command, value={}".format(steer))
-        else:
-            rospy.logdebug(
-                "Did no issue steer command, value={}, last value={}".format(steer, self.last_steer))
-
-        # self.current_velocity <= 0 to keep braking on traffic light
-        if abs(self.last_brake - brake) > EPSILON_BRAKE or self.current_velocity <= 0:
-            self.last_brake = brake
-            bcmd = BrakeCmd()
-            bcmd.enable = True
-            # braking only works with torque...
-            bcmd.pedal_cmd_type = BrakeCmd.CMD_TORQUE
-            bcmd.pedal_cmd = brake
-            self.brake_pub.publish(bcmd)
-            rospy.loginfo("Issued brake command, value={}".format(brake))
-        else:
-            rospy.logdebug(
-                "Did no issue brake command, value={}, last value={}".format(brake, self.last_brake))
+        self._controller.save()
+
+    def _dbw_enabled_cb(self, msg):
+        self._dbw_enabled = msg.data
+
+    def _tf_init_done_cb(self, msg):
+	rospy.logdebug("tf init done")
+	self._tf_init_done = msg
+
+    def _current_velocity_cb(self, msg):
+        self._current_velocity = msg
+
+    def _twist_cmd_cb(self, msg):
+        self._twist_cmd = msg
+
+    def _publish(self, throttle, brake, steer):
+        self._last_throttle = throttle
```

Comment:
I assume you can remove the `_last_throttle` and `_last_steer` fields as they are no longer used.